### PR TITLE
vesting: Allow retrieving the earned tokens during stopVesting

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: contracts/token/vesting/IERC20Vesting.sol [hunk 1]
```diff
@@ -65,6 +65,8 @@ interface IERC20Vesting {
     function transferVesting(address oldAddress, address newAddress) external;
 
     /// @dev Stops vesting for receiver and sends all tokens back to wallet
+    /// Any earned claimable amount is still claimable through `claim()`.
```
[the remark this answers is not in the document]
Comment (mijovic) [reply contracts/token/vesting/IERC20Vesting.sol:68]: I like using these, not sure if it is useful at all...

File: contracts/token/vesting/IERC20Vesting.sol [hunk 2]
```diff
@@ -65,6 +65,8 @@ interface IERC20Vesting {
...
+    /// Please note however that an account cannot be used again.
```
Comment (mijovic) [inline contracts/token/vesting/IERC20Vesting.sol:69]:
```suggestion
    /// @notice Account cannot be used again as the vesting contract receiver.
```

File: contracts/token/vesting/IERC20Vesting.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (axic) [reply contracts/token/vesting/IERC20Vesting.sol:68]: `@notice` vs `@dev` is misleading, see https://docs.soliditylang.org/en/v0.8.9/natspec-format.html?highlight=%40notice#tags

File: contracts/token/vesting/ERC20Vesting.sol [hunk 3]
```diff
@@ -86,13 +86,20 @@ contract ERC20Vesting is IERC20Vesting {
     function stopVesting(address receiver) external override onlyWallet {
         require(receiver != address(0), "Receiver cannot be 0.");
 
-        VestingTerms memory terms = vestingTerms[receiver];
+        VestingTerms storage terms = vestingTerms[receiver];
```
Comment (axic) [inline contracts/token/vesting/ERC20Vesting.sol:89]: This is the only "questionable" part. Using storage reference here to have the terms updated below. Alternatively could just explicitly put in the storage access.

Comment (mijovic) [reply contracts/token/vesting/ERC20Vesting.sol:89]: I don't mind it at all, like for me it is even better like this as it we reuse it in many different places

Comment (mijovic) [reply contracts/token/vesting/ERC20Vesting.sol:89]: I think we should use it as you did here

File: contracts/token/vesting/IERC20Vesting.sol (hunk 2 above)
Comment (mijovic) [reply contracts/token/vesting/IERC20Vesting.sol:69]: Can we add that account can't be used in vesting contract as receiver, `account cannot be used again` sound misleading

Comment (axic) [reply contracts/token/vesting/IERC20Vesting.sol]: Done.

**Review by mijovic — APPROVED** [review]

**Pull request merged**